kyo-prelude/shared/src/main/scala/kyo2/kernel/Safepoint.scala

```diff
@@ -120,25 +121,33 @@ object Safepoint:
                 finally self.exit(depth)
     end handle
 
-    private def handle(self: Safepoint, cause: Throwable): Nothing =
-        val size  = Math.min(self.traceIdx, maxTraceFrames)
-        val trace = copyTrace(self.trace, self.traceIdx).filter(_ != null).map(_.parse)
-        val toPad = trace.map(_.snippetShort.size).max + 1
-        val elements = trace.map { frame =>
-            StackTraceElement(
-                frame.snippetShort.reverse.padTo(toPad, ' ').reverse + " @ " + frame.declaringClass,
-                frame.methodName,
-                frame.position.fileName,
-                frame.position.lineNumber
+    // TODO compiler crash if private[kyo2]
+    def insertTrace(cause: Throwable)(using self: Safepoint): Unit =
```

Comment:
The compiler ends up generating invalid bytecode with this method marked as `private[kyo]` 😱 https://github.com/getkyo/kyo/actions/runs/9764789434/job/26955864828?pr=515#step:4:1988

Reply:
@fwbrasil perhaps we can mark all methods that are non-private (where they could/should be, but the compiler won't allow it) `@deprecated` with the reason `kyo.internal`. Then we should be able to silence that warning in our own code.

Reply:
Interesting! We have some other APIs that should also be private. For example, commands in effects like `Var` are public only because the effect type is public.